// @aws-amplify/core/lib/Platform/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageAction = exports.PushNotificationAction = exports.PubSubAction = exports.PredictionsAction = exports.InteractionsAction = exports.InAppMessagingAction = exports.GeoAction = exports.DataStoreAction = exports.AuthAction = exports.ApiAction = exports.AnalyticsAction = exports.Category = exports.Framework = void 0;
var Framework;
(function (Framework) {
    // < 100 - Web frameworks
    Framework["WebUnknown"] = "0";
    Framework["React"] = "1";
    Framework["NextJs"] = "2";
    Framework["Angular"] = "3";
    Framework["VueJs"] = "4";
    Framework["Nuxt"] = "5";
    Framework["Svelte"] = "6";
    // 100s - Server side frameworks
    Framework["ServerSideUnknown"] = "100";
    Framework["ReactSSR"] = "101";
    Framework["NextJsSSR"] = "102";
    Framework["AngularSSR"] = "103";
    Framework["VueJsSSR"] = "104";
    Framework["NuxtSSR"] = "105";
    Framework["SvelteSSR"] = "106";
    // 200s - Mobile framework
    Framework["ReactNative"] = "201";
    Framework["Expo"] = "202";
})(Framework = exports.Framework || (exports.Framework = {}));
var Category;
(function (Category) {
    Category["API"] = "api";
    Category["Auth"] = "auth";
    Category["Analytics"] = "analytics";
    Category["DataStore"] = "datastore";
    Category["Geo"] = "geo";
    Category["InAppMessaging"] = "inappmessaging";
    Category["Interactions"] = "interactions";
    Category["Predictions"] = "predictions";
    Category["PubSub"] = "pubsub";
    Category["PushNotification"] = "pushnotification";
    Category["Storage"] = "storage";
})(Category = exports.Category || (exports.Category = {}));
var AnalyticsAction;
(function (AnalyticsAction) {
    AnalyticsAction["Record"] = "1";
    AnalyticsAction["UpdateEndpoint"] = "2";
})(AnalyticsAction = exports.AnalyticsAction || (exports.AnalyticsAction = {}));
var ApiAction;
(function (ApiAction) {
    ApiAction["GraphQl"] = "1";
    ApiAction["Get"] = "2";
    ApiAction["Post"] = "3";
    ApiAction["Put"] = "4";
    ApiAction["Patch"] = "5";
    ApiAction["Del"] = "6";
    ApiAction["Head"] = "7";
})(ApiAction = exports.ApiAction || (exports.ApiAction = {}));
var AuthAction;
(function (AuthAction) {
    AuthAction["SignUp"] = "1";
    AuthAction["ConfirmSignUp"] = "2";
    AuthAction["ResendSignUp"] = "3";
    AuthAction["SignIn"] = "4";
    AuthAction["GetMFAOptions"] = "5";
    AuthAction["GetPreferredMFA"] = "6";
    AuthAction["SetPreferredMFA"] = "7";
    AuthAction["DisableSMS"] = "8";
    AuthAction["EnableSMS"] = "9";
    AuthAction["SetupTOTP"] = "10";
    AuthAction["VerifyTotpToken"] = "11";
    AuthAction["ConfirmSignIn"] = "12";
    AuthAction["CompleteNewPassword"] = "13";
    AuthAction["SendCustomChallengeAnswer"] = "14";
    AuthAction["DeleteUserAttributes"] = "15";
    AuthAction["DeleteUser"] = "16";
    AuthAction["UpdateUserAttributes"] = "17";
    AuthAction["UserAttributes"] = "18";
    AuthAction["CurrentUserPoolUser"] = "19";
    AuthAction["CurrentAuthenticatedUser"] = "20";
    AuthAction["CurrentSession"] = "21";
    AuthAction["VerifyUserAttribute"] = "22";
    AuthAction["VerifyUserAttributeSubmit"] = "23";
    AuthAction["VerifyCurrentUserAttribute"] = "24";
    AuthAction["VerifyCurrentUserAttributeSubmit"] = "25";
    AuthAction["SignOut"] = "26";
    AuthAction["ChangePassword"] = "27";
    AuthAction["ForgotPassword"] = "28";
    AuthAction["ForgotPasswordSubmit"] = "29";
    AuthAction["FederatedSignIn"] = "30";
    AuthAction["VerifiedContact"] = "31";
    AuthAction["UserSession"] = "32";
    AuthAction["CurrentUserCredentials"] = "33";
    AuthAction["CurrentCredentials"] = "34";
    AuthAction["CurrentUserInfo"] = "35";
    AuthAction["RememberDevice"] = "36";
    AuthAction["ForgetDevice"] = "37";
    AuthAction["FetchDevices"] = "38";
})(AuthAction = exports.AuthAction || (exports.AuthAction = {}));
var DataStoreAction;
(function (DataStoreAction) {
    DataStoreAction["Subscribe"] = "1";
    DataStoreAction["GraphQl"] = "2";
})(DataStoreAction = exports.DataStoreAction || (exports.DataStoreAction = {}));
var GeoAction;
(function (GeoAction) {
    GeoAction["SearchByText"] = "1";
    GeoAction["SearchForSuggestions"] = "2";
    GeoAction["SearchByPlaceId"] = "3";
    GeoAction["SearchByCoordinates"] = "4";
    GeoAction["SaveGeofences"] = "5";
    GeoAction["GetGeofence"] = "6";
    GeoAction["ListGeofences"] = "7";
    GeoAction["DeleteGeofences"] = "8";
})(GeoAction = exports.GeoAction || (exports.GeoAction = {}));
var InAppMessagingAction;
(function (InAppMessagingAction) {
    InAppMessagingAction["None"] = "0";
    InAppMessagingAction["SyncMessages"] = "1";
    InAppMessagingAction["IdentifyUser"] = "2";
})(InAppMessagingAction = exports.InAppMessagingAction || (exports.InAppMessagingAction = {}));
var InteractionsAction;
(function (InteractionsAction) {
    InteractionsAction["None"] = "0";
})(InteractionsAction = exports.InteractionsAction || (exports.InteractionsAction = {}));
var PredictionsAction;
(function (PredictionsAction) {
    PredictionsAction["Convert"] = "1";
    PredictionsAction["Identify"] = "2";
    PredictionsAction["Interpret"] = "3";
})(PredictionsAction = exports.PredictionsAction || (exports.PredictionsAction = {}));
var PubSubAction;
(function (PubSubAction) {
    PubSubAction["Subscribe"] = "1";
})(PubSubAction = exports.PubSubAction || (exports.PubSubAction = {}));
var PushNotificationAction;
(function (PushNotificationAction) {
    PushNotificationAction["None"] = "0";
})(PushNotificationAction = exports.PushNotificationAction || (exports.PushNotificationAction = {}));
var StorageAction;
(function (StorageAction) {
    StorageAction["Put"] = "1";
    StorageAction["Get"] = "2";
    StorageAction["List"] = "3";
    StorageAction["Copy"] = "4";
    StorageAction["Remove"] = "5";
    StorageAction["GetProperties"] = "6";
    StorageAction["Cancel"] = "7";
})(StorageAction = exports.StorageAction || (exports.StorageAction = {}));
